packages/components/src/tooltip/test/index.js

```diff
@@ -1,373 +1,323 @@
+		const user = userEvent.setup();
+		const onClickMock = jest.fn();
+
+		const { container } = render(
+			<Tooltip { ...props }>
+				<Button disabled onClick={ onClickMock }>
+					Click me
+				</Button>
+			</Tooltip>
+		);
+
+		// Note: this is testing for implementation details,
+		// but couldn't find a better way.
+		const buttonRect = screen
+			.getByRole( 'button', { name: 'Click me' } )
+			.getBoundingClientRect();
+		// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+		const eventCatcher = container.querySelector( '.event-catcher' );
+		await user.click( eventCatcher );
+		expect( onClickMock ).not.toHaveBeenCalled();
+
+		const eventCatcherRect = eventCatcher.getBoundingClientRect();
+		expect( buttonRect ).toEqual( eventCatcherRect );
+	} );
+
+	it( 'should not show tooltip if the mouse leaves the tooltip anchor before set delay', async () => {
+		const user = userEvent.setup();
+		const onMouseEnterMock = jest.fn();
+		const onMouseLeaveMock = jest.fn();
+		const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
+
+		render(
+			<>
+				<Tooltip { ...props }>
+					<Button
+						onMouseEnter={ onMouseEnterMock }
+						onMouseLeave={ onMouseLeaveMock }
+					>
+						Hover Me!
+					</Button>
 				</Tooltip>
-			);
-
-			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
-			const eventCatcher = container.querySelector( '.event-catcher' );
-			await user.click( eventCatcher );
-			expect( onClickMock ).not.toHaveBeenCalled();
-		} );
-
-		it( 'should not show tooltip if the mouse leaves the anchor before the tooltip has shown', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-
-			const MOUSE_LEAVE_DELAY = TOOLTIP_DELAY - 200;
-			const onMouseEnterMock = jest.fn();
-			const onMouseLeaveMock = jest.fn();
-
-			render(
-				<>
-					<Tooltip text="Help text">
-						<button
-							onMouseEnter={ onMouseEnterMock }
-							onMouseLeave={ onMouseLeaveMock }
-						>
-							Hover Me!
-						</button>
-					</Tooltip>
-					<button>Hover me instead!</button>
-				</>
-			);
-
-			const externalButton = screen.getByRole( 'button', {
-				name: 'Hover me instead!',
-			} );
-			const tooltipButton = screen.getByRole( 'button', {
+				<Button>Hover me instead!</Button>
+			</>
+		);
+
+		await user.hover(
+			screen.getByRole( 'button', {
 				name: 'Hover Me!',
-			} );
+			} )
+		);
 
-			await user.hover( tooltipButton );
+		// Tooltip hasn't appeared yet
+		expect( screen.queryByText( 'tooltip text' ) ).not.toBeInTheDocument();
+		expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
 
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
+		// Advance time by MOUSE_LEAVE_DELAY time
+		await new Promise( ( resolve ) =>
+			setTimeout( resolve, MOUSE_LEAVE_DELAY )
+		);
```

Comment:
This could be defined as a helper function since we use it multiple times. Totally optional, since it's small enough too.

Reply:
Thanks for pointing this out! Since it's just two instances, I'll leave it but it's definitely something I need to be mindful of in the future to avoid too much repetition.